### PR TITLE
fix: replace get primary key with convert row to key

### PR DESCRIPTION
**Pull request by connoratrug**
Replace use of sync utils.getPrimaryKey with async convertRowToPrimaryKey call.

getPrimaryKey did not handle the case where a key referred to a different schema,  convertRowToPrimaryKey does handle this case. 

- added convertRowToPrimaryKey call to the client for easier use from component ( no to pass schema in each call )

todo

- [x] fix issue with table molgenis , keys are resolved async therefore move rows into separate component and use vue component lifecycle and computed caching for resolving keys from rows

### CODE REVIEW

File: apps/molgenis-components/src/components/forms/InputRefItem.vue [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      isSelected: this.rowKey?.name === this.selection?.name,
+    };
+  },
+  computed: {
+    rowAsString() {
+      return flattenObject(this.rowKey).trim();
+    },
+    isSelected() {
+      return this.rowKey?.name === this.selection?.name;
+    },
```
Comment (chinook25) [inline apps/molgenis-components/src/components/forms/InputRefItem.vue:43]: Can't have this as both computed and in data. Just computed should be enough here I think.

**Review by chinook25 — CHANGES_REQUESTED** [review]
Component lib is throwing a lot of console warnings that are currently not happening om master

**Comment by connoratrug** [comment]
> Component lib is throwing a lot of console warnings that are currently not happening om master

can you give a example ?

File: apps/molgenis-components/src/components/forms/InputRefItem.vue (hunk 1 above)
Comment (connoratrug) [reply apps/molgenis-components/src/components/forms/InputRefItem.vue:42]: removed it from data

**Review by chinook25 — APPROVED** [review]

**Review by mswertz — APPROVED** [review]

**Pull request merged**